activerecord/lib/arel/visitors/to_sql.rb

```diff
@@ -618,6 +618,9 @@ def visit_Arel_Nodes_And(o, collector)
         end
 
         def visit_Arel_Nodes_Or(o, collector)
+          if o.inspect.include?("TableAlias")
+            o.right.instance_values["left"].relation = Nodes::TableAlias.new(o.right.instance_values["left"].relation, o.left.instance_values["left"].relation.right)
+          end
```

Comment:
Any code that rely on `inspect` and `instance_values` here isn't the right fix. It is probably that the right fix is to build the right AST instead of trying to checks is state when visiting it.

Reply:
Thanks for the feedback!  Looking into implementing it this way now.